File: 백준/Bronze/29614. 학점계산프로그램/학점계산프로그램.java
import java.util.HashMap;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        HashMap<String, Double> dic = new HashMap<>();
        dic.put("A", 4.0);
        dic.put("B", 3.0);
        dic.put("C", 2.0);
        dic.put("D", 1.0);
        dic.put("F", 0.0);
        dic.put("+", 0.5);

        String input = sc.nextLine();
        double dap = 0;
        int cnt = 0;
        for (int i = 0; i < input.length(); i++) {
            String grade = String.valueOf(input.charAt(i));
            dap += dic.get(grade);
            if (!grade.equals("+")) {
                cnt++;
            }
        }
        System.out.println(dap / cnt);
    }
}

File: 백준/Bronze/29614. 학점계산프로그램/학점계산프로그램.c
#include <stdio.h>
#include <string.h>

double get_score(char grade) {
    switch (grade) {
        case 'A': return 4.0;
        case 'B': return 3.0;
        case 'C': return 2.0;
        case 'D': return 1.0;
        case 'F': return 0.0;
        case '+': return 0.5;
        default: return 0.0;
    }
}

int main() {
    char input[1002];
    scanf("%s", input);
    double dap = 0.0;
    int cnt = 0;
    for (int i = 0; i < strlen(input); i++) {
        char grade = input[i];
        double score = get_score(grade);
        dap += score;
        if (grade != '+') {
            cnt++;
        }
    }
    printf("%.5f\n", dap / cnt);
    return 0;
}